src/test/java/seedu/address/logic/parser/AddressBookParserTest.java

```diff
@@ -84,21 +88,44 @@ public void parseCommand_exit() throws Exception {
     @Test
     public void parseCommand_find() throws Exception {
         List<String> keywords = Arrays.asList("foo", "bar", "baz");
-
-        FindCommand command = (FindCommand) parser.parseCommand(
-                FindCommand.COMMAND_WORD + " " + FindCommand.ARGUMENT_WORD + " "
-                        + keywords.stream().collect(Collectors.joining(" ")));
-        assertEquals(new FindCommand(new NameContainsKeywordsPredicate(keywords)), command);
-
-        FindEmployeeCommand employeeCommand = (FindEmployeeCommand) parser.parseCommand(
-                FindCommand.COMMAND_WORD + " " + FindEmployeeCommand.ARGUMENT_WORD + " "
-                        + keywords.stream().collect(Collectors.joining(" ")));
-        assertEquals(new FindEmployeeCommand(new NameContainsKeywordsPredicate(keywords)), employeeCommand);
-
-        FindPotentialCommand potentialCommand = (FindPotentialCommand) parser.parseCommand(
-                FindCommand.COMMAND_WORD + " " + FindPotentialCommand.ARGUMENT_WORD + " "
-                        + keywords.stream().collect(Collectors.joining(" ")));
-        assertEquals(new FindPotentialCommand(new NameContainsKeywordsPredicate(keywords)), potentialCommand);
+        NameContainsKeywordsPredicate nameContainsKeywordsPredicate = new NameContainsKeywordsPredicate(keywords);
+        PredicateContainer predicateContainer = new PredicateContainer()
+                .addNameContainsKeywordsPredicate(nameContainsKeywordsPredicate);
+
+        // FindCommand
+        FindCommand expected1 = new FindCommand(predicateContainer);
+        String input1 = FindCommand.COMMAND_WORD + " " + FindCommand.ARGUMENT_WORD + " " + PREFIX_NAME
+                + keywords.stream().collect(Collectors.joining(" "));
+        FindCommand actual1 = (FindCommand) parser.parseCommand(input1);
+        assertEquals(expected1, actual1);
+
+        // FindEmployeeCommand
+        FindEmployeeCommand expected2 = new FindEmployeeCommand(predicateContainer);
+        String input2 = FindCommand.COMMAND_WORD + " " + FindEmployeeCommand.ARGUMENT_WORD + " " + PREFIX_NAME
+                + keywords.stream().collect(Collectors.joining(" "));
+        FindEmployeeCommand actual2 = (FindEmployeeCommand) parser.parseCommand(input2);
+        assertEquals(expected2, actual2);
+
+        // FindPotentialCommand
+        FindPotentialCommand expected3 = new FindPotentialCommand(predicateContainer);
+        String input3 = FindCommand.COMMAND_WORD + " " + FindPotentialCommand.ARGUMENT_WORD + " " + PREFIX_NAME
+                + keywords.stream().collect(Collectors.joining(" "));
+        FindPotentialCommand actual3 = (FindPotentialCommand) parser.parseCommand(input3);
```

Comment:
I like the comprehensive testing for each Find class but perhaps instead of appending 1, 2, 3 or 4 to the names of the variable, there could be a better naming convention?